core/src/main/java/nl/nn/adapterframework/pipes/HashPipe.java

```diff
@@ -76,13 +76,13 @@ public PipeRunResult doPipe (Message message, PipeLineSession session) throws Pi
 			ParameterList parameterList = getParameterList();
 			ParameterValueList pvl = parameterList==null ? null : parameterList.getValues(message, session);
 			if(pvl != null) {
-				String authAliasParamValue = (String)pvl.getValue("authAlias");
-				if (StringUtils.isNotEmpty(authAliasParamValue)) {
-					authAlias = authAliasParamValue;
+				ParameterValue authAliasParamValue = pvl.get("authAlias");
+				if (authAliasParamValue != null) {
+					authAlias = authAliasParamValue.asStringValue();
 				}
-				String secretParamValue = (String)pvl.getValue("secret");
-				if (StringUtils.isNotEmpty(secretParamValue)) {
-					secret = secretParamValue;
+				ParameterValue secretParamValue = pvl.get("secret");
+				if (secretParamValue != null) {
+					secret = secretParamValue.asStringValue();
```

Comment:
Nou moet je niet zomaar deze isNotEmpies door != null vervangen

Reply:
!= null betekend bestaat niet, een parameter is nooit null of leeg, hij krijgt de waarde van de message als hij zelf geen waarde heeft..

Reply:
En als de message dan leeg is?

Reply:
Zie HashPipeTest regel 158

Reply:
daar is de message toch niet leeg?